include/seahorn/Houdini.hh

```diff
@@ -0,0 +1,68 @@
+#ifndef HOUDINI__HH_
+#define HOUDINI__HH_
+
+#include "llvm/Pass.h"
+#include "llvm/IR/Module.h"
+#include "seahorn/HornClauseDB.hh"
+#include "seahorn/HornifyModule.hh"
+
+#include "ufo/Expr.hpp"
+#include "ufo/Smt/Z3n.hpp"
+#include "ufo/Smt/EZ3.hh"
+
+namespace seahorn
+{
+  using namespace llvm;
+
+  class Houdini : public llvm::ModulePass
+  {
+  public:
+    static char ID;
+
+    Houdini() : ModulePass(ID) {}
+    virtual ~Houdini() {}
+
+    virtual bool runOnModule (Module &M);
+    virtual void getAnalysisUsage (AnalysisUsage &AU) const;
+    virtual const char* getPassName () const {return "Houdini";}
+
+    void printDB (const HornClauseDB &db);
+    void printHello ();
+    Expr guessCandidate(Expr pred);
+    int testGuessCandidate(Expr pred);
+    bool validateRule(Expr cand_app, HornifyModule &hm);
+    void workListAlgo(HornClauseDB &db, HornifyModule &hm);
+
+    template<typename OutputIterator>
```

Comment:
Please break the methods into public and private. 
Move as much out of the header file as you can (especially support functions and structs such as `IsBVar`)